coxeter/shape_classes/circle.py

```
@@ -89,11 +89,19 @@ def perimeter(self):
         """float: Get the perimeter of the circle."""
         return 2 * np.pi * self.radius
 
+    @perimeter.setter
+    def perimeter(self, value):
+        self._radius = value / (2 * np.pi)
```

We need to do some amount of validation of the input here. The division will automatically fail if users pass something completely bogus like `circle.perimeter = 'hello tobias'`, but if you set a perimeter <= 0 you won't currently warn the user that they're doing something invalid. I would raise an error in such cases.

---

added checks.